storage/repository/errors.go

```diff
@@ -15,3 +15,18 @@ const OperationNotSupportedError operationNotSupportedError = operationNotSuppor
 func (e operationNotSupportedError) Error() string {
 	return string(OperationNotSupportedError)
 }
+
+// Declare conformance with Error interface
+var _ error = URIRootError
+
+type uriRootError string
+
+// URIRootError should be thrown by fyne.URI implementations when the caller
+// attempts to take the parent of the root. This way, downstream code that
+// wants to programmatically walk up a URIs parent's will know when to stop
+// iterating.
+const URIRootError uriRootError = uriRootError("Cannot take the parent of the root element in a URI")
```

Comment:
Was this new for this release? If so, it needs a Since annotation. If not then we shouldn't move it without appropriate deprecations

Reply:
a8a4855ec9b5cc522c7aaed202c8e13353699abd